app/test_main.py

```diff
@@ -1 +1,38 @@
-# write your code here
+from typing import Any
+
+import pytest
+
+from app.main import can_access_google_page
+
+
+@pytest.mark.parametrize(
+    "internet,valid_url,expected",
+    [
+        (True, True, "Accessible"),
+        (True, False, "Not accessible"),
+        (False, True, "Not accessible"),
+        (False, False, "Not accessible"),
+    ],
+)
+def test_access_google_page(
+        monkeypatch: Any,
+        internet: bool,
+        valid_url: bool,
+        expected: Any
+) -> None:
+    def mock_has_internet_connection() -> bool:
+        return internet
+
+    def mock_valid_google_url(url: str) -> bool:
+        return valid_url
```

Comment:
Use `mock.patch` to create mocks for `valid_google_url` and `has_internet_connection`